pretyping/constr_matching.ml

```diff
@@ -296,38 +298,46 @@ let matches_core env sigma convert allow_partial_app allow_bound_rels
             if not ci1.cip_extensible && not (Int.equal (List.length br1) n2)
             then raise PatternMatchingFailure
           in
-	  let chk_branch subst (j,n,c) =
+	  let chk_branch current (j,n,c) =
 	    (* (ind,j+1) is normally known to be a correct constructor
 	       and br2 a correct match over the same inductive *)
 	    assert (j < n2);
-	    sorec ctx env subst c br2.(j)
+	    sorec ctx env current c br2.(j)
 	  in
-	  let chk_head = sorec ctx env (sorec ctx env subst a1 a2) p1 p2 in
+	  let chk_head = sorec ctx env (sorec ctx env current a1 a2) p1 p2 in
 	  List.fold_left chk_branch chk_head br1
 
-      |	PFix c1, Fix _ when eq_constr (mkFix c1) cT -> subst
-      |	PCoFix c1, CoFix _ when eq_constr (mkCoFix c1) cT -> subst
+      |	PFix c1, Fix _ when eq_constr (mkFix c1) cT -> current
```

Comment:
I know that nobody actually uses pattern-matching over (co)fixpoints because it's just going to fail anyway, but for the sake of consistency, we could use equality of terms up to universes here as well.